packaging/packaging_test.go

```diff
@@ -0,0 +1,782 @@
+	}
+}
+
+func TestNeedSplit(t *testing.T) {
+	// create the needSplitTests
+	needSplitTests := []struct {
+		name     string
+		fileList []os.FileInfo
+		maxBytes int64
+		want     bool
+	}{
+		{
+			name:     "test split required",
+			fileList: testDirs.large.files,
+			maxBytes: 1 * 1024 * 1024,
+			want:     true,
+		},
+		{
+			name:     "test split not required",
+			fileList: testDirs.small.files,
+			maxBytes: 100 * 1024 * 1024,
+			want:     false,
+		},
+	}
+	for _, tt := range needSplitTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			testPackager.maxBytes = tt.maxBytes
+			got := testPackager.needSplit(tt.fileList)
+			if tt.want != got {
+				t.Errorf("Outcome for test %s:\nReceived: %v\nExpected: %v", tt.name, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBuildLocalCSVFileList(t *testing.T) {
+	// create the buildLocalCSVFileList tests
+	buildLocalCSVFileListTests := []struct {
+		name     string
+		dirName  string
+		fileList []os.FileInfo
+	}{
+		{
+			name:     "test regular dir",
+			dirName:  filepath.Join(testDirs.large.directory, "data"),
+			fileList: testDirs.large.files,
+		},
+		{
+			name:     "test empty dir",
+			dirName:  filepath.Join(testDirs.empty.directory, "data"),
+			fileList: testDirs.empty.files,
+		},
+	}
+	for _, tt := range buildLocalCSVFileListTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			got := testPackager.buildLocalCSVFileList(tt.fileList, tt.dirName)
+			want := make(map[int]string)
+			fileInfoList, _ := ioutil.ReadDir(tt.dirName)
+			for idx, file := range fileInfoList {
+				// generate the expected file list
+				if strings.HasSuffix(file.Name(), ".csv") {
+					want[idx] = filepath.Join(tt.dirName, file.Name())
+				}
+			}
+			for index, fileName := range want {
+				if fileName != got[index] {
+					t.Errorf("Expected %s but got %s", fileName, got[index])
+				}
+			}
+
+		})
+	}
+}
+
+func TestMoveFiles(t *testing.T) {
+	// create the moveFiles tests
+	moveFilesTests := []struct {
+		name     string
+		dirName  string
+		fileUUID string
+		want     []os.FileInfo
+	}{
+		{
+			name:     "test moving dir",
+			dirName:  testDirs.moving.directory,
+			fileUUID: uuid.New().String(),
+			want:     testDirs.large.files,
+		},
+		{
+			name:     "test empty dir",
+			dirName:  testDirs.empty.directory,
+			fileUUID: uuid.New().String(),
+			want:     nil,
+		},
+		{
+			name:     "test restricted dir",
+			dirName:  testDirs.restricted.directory,
+			fileUUID: uuid.New().String(),
+			want:     nil,
+		},
+	}
+	for _, tt := range moveFilesTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			dirconfig.ParentDir = tt.dirName
+			dirCfg.GetDirectoryConfig()
+			testPackager.DirCfg = dirCfg
+			testPackager.uid = tt.fileUUID
+			got, _ := testPackager.moveFiles()
+			if tt.want != nil {
```

Comment:
The logic flow is a little difficult to understand here. It would be good to consolidate conditions:
```
if tt.want == nil && got != nil {
	t.Errorf("expected moved files to be nil")
}
if tt.want != nil && got == nil {
	t.Errorf("expected moved files to be not nil")
}
```
then do the for loop:
```
if tt.want != nil && got != nil {
...
}
```

Reply:
I re-arranged the logic as you suggested to make it easier to follow.